harness/harness.go

```diff
@@ -464,14 +489,24 @@ func (t *H) report() {
 	}
 	dstr := fmtDuration(t.duration)
 	format := "--- %s: %s (%s)\n"
+
+	var status string
 	if t.Failed() {
-		t.flushToParent(format, "FAIL", t.name, dstr)
-	} else if t.suite.opts.Verbose {
-		if t.Skipped() {
-			t.flushToParent(format, "SKIP", t.name, dstr)
-		} else {
-			t.flushToParent(format, "PASS", t.name, dstr)
-		}
+		status = "FAIL"
+		t.flushToParent(format, status, t.name, dstr)
+	} else if t.Skipped() {
+		status = "SKIP"
+	} else {
+		status = "PASS"
+	}
+
+	if t.suite.opts.Verbose {
+		t.flushToParent(format, status, t.name, dstr)
+	} else {
+		t.mu.Lock()
+		defer t.mu.Unlock()
+		b := t.output.Bytes()
+		t.publishToReporters(t.name, status, t.duration, b)
```

Comment:
`publishToReporters` being called both here and in `flushToParent` seems awkward.

Couldn't we call it only once and have it deal with output handling itself?

As is, having to realize that `flushToParent` is implicitly calling it makes this code harder to follow and seems like a possible footgun.